### PR TITLE
Dont show cancelled comps in my past comps table

### PR DESCRIPTION
**Pull request by DanielEgdal**
From the original email, this solves https://github.com/thewca/worldcubeassociation.org/issues/7597

### CODE REVIEW

File: WcaOnRails/app/controllers/competitions_controller.rb [hunk 1]
```diff
@@ -588,7 +588,7 @@ def my_competitions
       competition_ids.concat(current_user.person.competitions.pluck(:competitionId))
     end
     competitions = Competition.includes(:delegate_report, :delegates)
-                              .where(id: competition_ids.uniq)
+                              .where(id: competition_ids.uniq).where("cancelled_at is null or end_date >= curdate()")
```
Comment (FinnIckler) [inline WcaOnRails/app/controllers/competitions_controller.rb:591]: why do we need the `end_date >= curdate()` check here?

Comment (DanielEgdal) [reply WcaOnRails/app/controllers/competitions_controller.rb:591]: The logic is that an organiser might still have duties to perform for a competition until the date of the competition has passed. For example, mailing all competitors about the cancellation. In general ensuring ease of access until it is certain that they won't need to frequently visit the page anymore.

Comment (FinnIckler) [reply WcaOnRails/app/controllers/competitions_controller.rb:591]: ah ok, can you write a comment about that?

Comment (DanielEgdal) [reply WcaOnRails/app/controllers/competitions_controller.rb:594]: Done

**Review by FinnIckler — APPROVED** [review]
LGTM!

**Pull request merged**